### PR TITLE
search: alert event for streaming PoC

### PR DESCRIPTION
**Pull request by keegancsmith**
We exactly match the how it is done in GQL. We will likely want to
adjust this once we start implementing the designs for streaming search.

### CODE REVIEW

File: client/web/src/search/stream.tsx [hunk 1]
```diff
@@ -69,6 +70,17 @@ interface Filter {
     kind: string
 }
 
+interface Alert {
```
Comment (limitedmage) [inline client/web/src/search/stream.tsx:73]: I'm concerned that adding all these typed interfaces here is not going to scale as we add more functionality. Could we generate typings from the Go code? Or at least extract these into another file (eg. `streamTypes.d.ts`)

File: client/web/src/search/stream.tsx [hunk 2]
```diff
@@ -189,6 +201,18 @@ const toGQLSearchFilter = (filter: Omit<Filter, 'type'>): GQL.ISearchFilter => (
...
 })
 
+const toGQLSearchAlert = (alert: Alert): GQL.ISearchAlert => ({
+    __typename: 'SearchAlert',
+    title: alert.title,
+    description: alert.description || null,
+    proposedQueries:
+        alert.proposedQueries?.map(pq => ({
+            __typename: 'SearchQueryDescription',
+            description: pq.description ? pq.description : null,
```
Comment (limitedmage) [inline client/web/src/search/stream.tsx:211]:
```suggestion
            description: pq.description || null,
```

**Review by limitedmage — APPROVED** [review]

File: client/web/src/search/stream.tsx (hunk 1 above)
Comment (keegancsmith) [reply client/web/src/search/stream.tsx:73]: Purely for the POC. Being explicit to help scope out the API. We can play with more flexible ways to do it as the need  arises. Personally I'd prefer to keep it simple and just do a little bit of duplicate work. If we find we are doing it too often, maybe its a sign we need a better API design?

File: client/web/src/search/stream.tsx (hunk 2 above)
Comment (keegancsmith) [reply client/web/src/search/stream.tsx:211]: weird when I originally wrote this I did that and eslint/tsc shouted at me. I likely had some other issues that confused it, cause it seems to work now.

**Pull request merged**